dfsclient/src/main/java/org/hpccsystems/dfs/client/BinaryRecordReader.java

```diff
@@ -312,8 +325,7 @@ private Object parseFlatField(FieldDef fd, boolean isLittleEndian) throws Unpars
                     intValue = getUnsigned((int) fd.getDataLen(), fd.getSourceType() == HpccSrcType.LITTLE_ENDIAN);
                     if (intValue < 0)
                     {
-                        log.warn(
-                                "Detected possible unsigned value overflow in field '" + fd.getFieldName() + "'. Ensure proper value interpretation");
+                        messages.addMessage("Detected possible unsigned value overflow, ensure proper value interpretation",  MessageType.Warning, fd.getFieldName());
```

Comment:
Referencing the comment below, would it be possible to pre-calculate this error key for each field so that we aren't incurring the createKey() cost for each messsage?